docs/sources/developers/plugins/migration-guide.md

```diff
@@ -0,0 +1,151 @@
+
+class MyPanelCtrl extends MetricsPanelCtrl {
+  // ...
+}
+
+export { MyPanelCtrl as PanelCtrl };
+```
+
+In >=7.0, plugins now export a PanelPlugin from module.ts
+
+```ts
+import { PanelPlugin } from '@grafana/data';
+
+export const plugin = new PanelPlugin<MyOptions>(MyPanel);
+```
+
+where MyPanel is a React component containing the props from PanelProps.
+
+```ts
+import { PanelProps } from '@grafana/data';
+
+interface Props extends PanelProps<SimpleOptions> {}
+
+export const MyPanel: React.FC<Props> = ({ options, data, width, height }) => {
+  // ...
+}
+```
+
+### Migrate a data source plugin
+
+For many data sources, the `ConfigEditor` is the simplest component, so it’s a good candidate to start migrating.
+
+1. When you’ve defined your configuration model, the next step is to implement the `testDatasource()` method on the class that extends `DataSourceApi`.
+1. Once you know that you can successfully configure and access the external API, a good next step is to implement the `query()` method. At this point, you can hard-code your query, because we haven’t yet implemented the query editor component.
+1. The `query()` method supports both the new data frame response and the old TimeSeries response, so don’t worry about converting to the new format just yet.
+1. Next, implement the `QueryEditor`. How much work this requires depends on how complex your query model is.
+
+By now, you should be able to release your new version.
+
+To fully migrate to the new plugin platform, convert the time series response to a data frame response.
+
+### Migrate to data frames
+
+In <7.0, data source and panel plugins exchanged data using either time series or tables. In >=7.0 plugins use the new Data Frame format to pass data from data sources to panels.
+
+Grafana 7 is backward compatible with the old data format used in previous versions. Panels and data sources using the old format will still work with plugins using the new data frame format.
+
+The `DataQueryResponse` returned by the `query` method can be either a [LegacyResponseData](https://grafana.com/docs/grafana/latest/packages_api/data/legacyresponsedata/) or a [DataFrame](https://grafana.com/docs/grafana/latest/packages_api/data/dataframe/).
+
+The [toDataFrame()](https://grafana.com/docs/grafana/latest/packages_api/data/todataframe/) function converts a legacy response, such as `TimeSeries` or `Table`, to a `DataFrame`. Use it to gradually move your code to the new format.
+
+```ts
+import { toDataFrame } from '@grafana/data';
+```
+
+```ts
+async query(options: DataQueryRequest<MyQuery>): Promise<DataQueryResponse> {
+  return {
+    data: options.targets.map(query => {
+      const timeSeries: TimeSeries = await doLegacyRequest(query);
+      return toDataFrame(timeSeries);
+    }
+  };
+}
+```
+
+For more information, refer to [Data frames](https://grafana.com/docs/grafana/latest/developers/plugins/data-frames/).
+
+## Troubleshooting
```

Comment:
Avoid stacked headings. Add at least a one-sentence intro to tell the user what to expect in this section.

Also, please make the heading more specific, like "Troubleshoot migration." I'm thinking of all the search results called "Troubleshooting" that will come up if we have multiple "Troubleshooting" headings in the docs.